### DISCUSSION TITLE
What is the recommended way to calculate FLOP?

### DISCUSSION THREAD

**Discussion by zenineasa** [coin-or/Ipopt#669] [Q&A] [answered]
I have been looking to see if there is a way to extract information on the number of floating point operations taken to solve a problem. Is there a direct way to do that in IPOPT? If otherwise, is there a recommended method to do the same?

**Comment by svigerske** [comment, marked as answer]
There is no direct way to do that. Ipopt itself doesn't count this. You would have to rely on the CPU to do that for you. A tool like per may help to get the counts for a process:
- https://bnikolic.co.uk/blog/hpc-howto-measure-flops
- https://bnikolic.co.uk/blog/python/flops/2019/09/27/python-counting-events.html
- https://stackoverflow.com/questions/4335011/measuring-flops-of-an-application-with-the-linux-perf-tool
